### PR TITLE
Show recipient email in sent mailboxes

### LINKED ISSUE #7627: Sent folder should display the recipient by default, not the sender (it's always me !)
### Is your feature request related to a problem? Please describe.

The Sent folder is useless because it only shows the sender of all emails, and... it's always the owner of the box

### Describe the solution you'd like

By default, the Sent folder should display the recipient(s) of the email.

### Describe alternatives you've considered

Clicking on each mail to see the recipient...

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hamza221**
Fixes #7627

**reproduction:**

1. add an imap account to Mail
2. create a new mailbox "test"
3. set "test" as sent mailbox in the account settings
4. send an email to someone else
5. open "test"

-> it shows your name of the message

**expected behaviour:** it shows the recipient

#Signed-off-by: hamza221 <hamzamahjoubi221@gmail.com>

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Comment by hamza221** [comment]
Fixes #7995

**Comment by hamza221** [comment]
Fixes #7627

**Review by GretaD — APPROVED** [review]
tested and works

**Comment by GretaD** [comment]
ahhh, and of course test need adjustment -_-. 

Let me know if you need help with that as i am familiar with that test case

File: src/components/Envelope.vue [hunk 1]
```diff
@@ -400,7 +400,7 @@ export default {
 		},
 		addresses() {
 			// Show recipients' label/address in a sent mailbox
-			if (this.mailbox.specialRole === 'sent') {
+			if (this.mailbox.specialRole === 'sent' || this.mailbox.sentMailboxId === this.account.databaseId) {
```
Comment (kesselb) [inline src/components/Envelope.vue:403]:
```suggestion
			if (this.mailbox.specialRole === 'sent' || this.account.sentMailboxId === this.mailbox.databaseId) {
```

Comment (hamza221) [reply src/components/Envelope.vue]: You're right, thank you for catching it :)

Comment (kesselb) [reply src/components/Envelope.vue]: You are welcome

**Comment by kesselb** [comment]
I guess the tests are failing because the component state is not mocked properly. 
It's unrelated to your patch but for a happy CI we need to fix it :see_no_evil: 

https://github.com/nextcloud/mail/blob/be8577d0e5e4562af2f6b47b151aa7bd17c108d3/src/tests/unit/components/Envelope.vue.spec.js#L87-L111

account.sentMailboxId and mailbox.databaseId are both undefined. 
undefined === undefined is true :laughing: 

Please add sentMailboxId to account and databaseId to mailbox (in Envelope.vue.spec.js) to fix the tests. 
In some tests cases we have two mailbox objects. That's a mistake and we should merge them. 

Thank you :+1:

**Pull request merged**